### ISSUE TITLE
Optimal Hard Negative Embeddings Integrate into recognition and retrieval tasks

### ISSUE THREAD

**Issue by syh-hue** [puneesh00/LoOp#1]
Hi, thank you for your excellent work. I want to learn your ideas and integrate them into my project. Add "Optimal Hard Negative Embeddings" to ml_loss.py?
How do I add Optimal Hard Negative Embeddings to my task, currently my Reid task includes triplet loss, softmax, and circle loss.
There is also mxnet. Do I need to rewrite ml_loss.py for my tasks in pytorch?
Looking forward for your response, thanks!
